sympy/functions/special/bessel.py

```diff
@@ -298,11 +308,32 @@ def _eval_rewrite_as_besseli(self, nu, z, **kwargs):
     def _eval_rewrite_as_yn(self, nu, z, **kwargs):
         return sqrt(2*z/pi) * yn(nu - S.Half, self.argument)
 
+    def _eval_as_leading_term(self, x, cdir=0):
+        nu, z = self.args
+        arg = z.as_leading_term(x)
+        if x in arg.free_symbols:
+            return z**nu
+        else:
+            return self.func(*self.args)
+
     def _eval_is_extended_real(self):
         nu, z = self.args
         if nu.is_integer and z.is_positive:
             return True
 
+    def _eval_nseries(self, x, n, logx, cdir=0):
+        nu, z = self.args
+
+        if z.limit(x, 0) is nu is S.Zero:
+            k = Symbol('k')
+            term = (-1)**(k + 1) * harmonic(k) * (z / 2)**(2*k) / (factorial(k)**2)
+            s = [term.subs(k, j) for j in range(0, n) if 2 * j < n]
```

Comment:
The term expression could be included in the list comprehension with something like `for k in range((n+1)//2)`. A new symbol need not be created.

Reply:
OK I will do that. In the original PR, it was mentioned to break up the term calculation like this to make it easier to debug.

Reply:
I agree that a fixed range together with a condition like `2 * j < n` is more readable than a computed range, but it seems that a computation is needed in general. For example, if `z = cbrt(x)` then a range of ca. 3n/2 terms is necessary.

Reply:
`z = cbrt(x)` or even `sqrt(x)` was leading to infinite recursion due to besselj series not being implemented. It is now implemented and fixed